packages/odyssey-react-mui/src/ui-shell/useMatchAppElementToUiShellAppArea.ts

```diff
@@ -0,0 +1,189 @@
+   */
+  uiShellAppAreaRef: RefObject<HTMLDivElement>;
+  /**
+   * Array of refs of items that indirectly resize the app content area such as "side nav" and "top nav".
+   */
+  uiShellResizableRefs: Array<RefObject<HTMLDivElement>>;
+} & Pick<UiShellContentProps, "appElement" | "appElementScrollingMode">;
+
+export const useMatchAppElementToUiShellAppArea = ({
+  appElement,
+  appElementScrollingMode,
+  paddingMode,
+  uiShellAppAreaRef,
+  uiShellResizableRefs,
+}: UseMatchAppElementToUiShellAppAreaProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const parentContainerRef = useRef<HTMLDivElement>(null);
+
+  const appContainerElementStyles = useMemo<CSSProperties>(
+    () => ({
+      ...(paddingMode === "comfortable"
+        ? {
+            paddingBlock: odysseyDesignTokens.Spacing5,
+            paddingInline: odysseyDesignTokens.Spacing8,
+          }
+        : {}),
+      ...(paddingMode === "compact"
+        ? {
+            paddingBlock: odysseyDesignTokens.Spacing5,
+            paddingInline: odysseyDesignTokens.Spacing5,
+          }
+        : {}),
+      ...(appElementScrollingMode === "horizontal" ||
+      appElementScrollingMode === "both"
+        ? {
+            overflowX: "auto",
+          }
+        : {
+            overflowX: "hidden",
+          }),
+      ...(appElementScrollingMode === "vertical" ||
+      appElementScrollingMode === "both"
+        ? {
+            overflowY: "auto",
+          }
+        : {
+            overflowY: "hidden",
+          }),
+    }),
+    [appElementScrollingMode, paddingMode, odysseyDesignTokens],
+  );
+
+  useEffect(() => {
+    // Once `appContainerRef` is rendered, we can position `appContainerElement` on top to match.
+    if (uiShellAppAreaRef.current && appElement && parentContainerRef.current) {
+      let animationFrameId: number;
+
+      const updateStyles = () => {
+        cancelAnimationFrame(animationFrameId);
+
+        animationFrameId = requestAnimationFrame(() => {
+          if (uiShellAppAreaRef.current && parentContainerRef.current) {
+            setStylesToMatchElement({
+              additionalStyles: appContainerElementStyles,
+              appContentReferenceElement: uiShellAppAreaRef.current,
+              appContainerElement: appElement,
+              parentElement: parentContainerRef.current,
+            });
+          }
+        });
+      };
+
+      // These refs might change by the time we unsubscribe, so we need to keep references to the original elements.
+      const resizingElements = uiShellResizableRefs
+        .map((resizingRef) => resizingRef.current)
+        .filter((element): element is NonNullable<typeof element> =>
+          Boolean(element),
+        );
+
+      // Set up a mutation observer to sync later updates
+      const observer = new ResizeObserver(updateStyles);
+
+      // document.addEventListener("ready", updateStyles);
```

Comment:
nit there's a couple of commented out code here -- should we keep them or remove them?